File: frontend/src/api/sales.ts
/**
 * 销售数据 API
 */
import { apiClient } from '@/utils/request'

export interface SalesData {
  id: number
  product_id: number
  product_name?: string
  date: string
  views: number
  clicks: number
  orders: number
  sales_amount: number
  created_at: string
}

export interface SalesDataCreateParams {
  product_id: number
  date: string
  views?: number
  clicks?: number
  orders?: number
  sales_amount?: number
}

export interface SalesDataUpdateParams {
  views?: number
  clicks?: number
  orders?: number
  sales_amount?: number
}

export interface SalesDataListParams {
  page?: number
  page_size?: number
  product_id?: number
  start_date?: string
  end_date?: string
}

export interface SalesDataListResponse {
  items: SalesData[]
  total: number
  page: number
  page_size: number
}

export interface SalesTrendData {
  date: string
  views: number
  clicks: number
  orders: number
  sales_amount: number
}

export interface SalesStatistics {
  total_views: number
  total_clicks: number
  total_orders: number
  total_sales_amount: number
  avg_conversion_rate: number
  trend_data: SalesTrendData[]
}

export interface ProductSalesRanking {
  product_id: number
  product_name: string
  total_orders: number
  total_sales_amount: number
  total_views: number
  conversion_rate: number
}

export interface SalesAnalyticsResponse {
  statistics: SalesStatistics
  top_products: ProductSalesRanking[]
  period_start: string
  period_end: string
}

/**
 * 创建销售数据
 */
export const createSalesData = (data: SalesDataCreateParams) => {
  return apiClient.post<SalesData>('/api/v1/sales/create', data)
}

/**
 * 获取销售数据列表
 */
export const getSalesDataList = (params?: SalesDataListParams) => {
  return apiClient.get<SalesDataListResponse>('/api/v1/sales/list', { params })
}

/**
 * 更新销售数据
 */
export const updateSalesData = (id: number, data: SalesDataUpdateParams) => {
  return apiClient.put<SalesData>(`/api/v1/sales/${id}`, data)
}

/**
 * 删除销售数据
 */
export const deleteSalesData = (id: number) => {
  return apiClient.delete(`/api/v1/sales/${id}`)
}

/**
 * 获取销售分析数据
 */
export const getSalesAnalytics = (days: number = 30, product_id?: number) => {
  return apiClient.get<SalesAnalyticsResponse>('/api/v1/sales/analytics', {
    params: { days, product_id },
  })
}

File: backend/app/schemas/sales.py
"""
销售数据相关的 Schema
"""
from typing import Optional, List
from datetime import date as DateType, datetime
from pydantic import BaseModel, Field


class SalesDataCreateRequest(BaseModel):
    """创建销售数据请求"""
    product_id: int = Field(..., description="产品ID")
    date: DateType = Field(..., description="日期")
    views: int = Field(default=0, description="浏览量")
    clicks: int = Field(default=0, description="点击量")
    orders: int = Field(default=0, description="订单量")
    sales_amount: float = Field(default=0.0, description="销售额")


class SalesDataUpdateRequest(BaseModel):
    """更新销售数据请求"""
    views: Optional[int] = Field(default=None, description="浏览量")
    clicks: Optional[int] = Field(default=None, description="点击量")
    orders: Optional[int] = Field(default=None, description="订单量")
    sales_amount: Optional[float] = Field(default=None, description="销售额")


class SalesDataResponse(BaseModel):
    """销售数据响应"""
    id: int
    product_id: int
    product_name: Optional[str] = None  # 关联查询
    date: DateType
    views: int
    clicks: int
    orders: int
    sales_amount: float
    created_at: datetime

    class Config:
        from_attributes = True


class SalesDataListResponse(BaseModel):
    """销售数据列表响应"""
    items: List[SalesDataResponse]
    total: int
    page: int
    page_size: int


class SalesTrendData(BaseModel):
    """销售趋势数据"""
    date: str
    views: int
    clicks: int
    orders: int
    sales_amount: float


class SalesStatistics(BaseModel):
    """销售统计数据"""
    total_views: int
    total_clicks: int
    total_orders: int
    total_sales_amount: float
    avg_conversion_rate: float  # 平均转化率
    trend_data: List[SalesTrendData]


class ProductSalesRanking(BaseModel):
    """产品销售排行"""
    product_id: int
    product_name: str
    total_orders: int
    total_sales_amount: float
    total_views: int
    conversion_rate: float


class SalesAnalyticsResponse(BaseModel):
    """销售分析响应"""
    statistics: SalesStatistics
    top_products: List[ProductSalesRanking]
    period_start: DateType
    period_end: DateType
